conwayste/src/client.rs

```diff
@@ -544,34 +410,97 @@ impl EventHandler for MainState {
     fn update(&mut self, ctx: &mut Context) -> GameResult<()> {
         let duration = timer::duration_to_f64(timer::delta(ctx)); // seconds
 
-        self.receive_net_updates();
+        self.receive_net_updates(ctx)?;
+
+        let current_screen = match self.screen_stack.last() {
+            Some(screen) => screen,
+            None => panic!("Error in main thread update! Screen_stack is empty!"),
+        };
 
-        match self.stage {
-            Stage::Intro(mut remaining) => {
+        match current_screen {
+            Screen::Intro => {
+                // Any key should skip the intro
+                if self.inputs.key_info.key.is_some() || (self.power_on_timestamp > INTRO_DURATION) {
+                    self.screen_stack.pop();
+                    self.screen_stack.push(Screen::Menu);
+                } else {
+                    self.power_on_timestamp += duration;
```

Comment:
I expect this will cause rounding errors to accumulate.

Reply:
How so? They're both `f64`, and I'm not too worried about precision here.

Reply:
OK, fair enough!